scheduling/non_preemptive_shortest_job_first.py

```diff
@@ -0,0 +1,116 @@
+# SJF
+"""
+Non-preemptive Shortest Job First
+Shortest execution time process is chosen for the next execution.
+https://www.guru99.com/shortest-job-first-sjf-scheduling.html
+https://en.wikipedia.org/wiki/Shortest_job_next
+"""
+
+
+from __future__ import annotations
+
+from statistics import mean
+
+
+def calculate_waitingtime(
+    arrival_time: list[int], burst_time: list[int], no_of_processes: int
+) -> list[int]:
+
+    """
+    Calculate the waiting time of each processes
+
+    Return: The waiting time for each process.
+    >>> calculate_waitingtime([0,1,2], [10, 5, 8], 3)
+    [0, 9, 13]
+    >>> calculate_waitingtime([1,2,2,4], [4, 6, 3, 1], 4)
+    [0, 7, 4, 1]
+    >>> calculate_waitingtime([0,0,0], [12, 2, 10],3)
+    [12, 0, 2]
+    """
+
+    waiting_time = [0] * no_of_processes
+    remaining_time = [0] * no_of_processes
+
+    """
+    Initialize remaining_time to waiting_time.
+    """
```

Comment:
You might want to use `#`-comments in the middle of the functions.
```suggestion
    # Initialize remaining_time to waiting_time.
```